src/uct/ib/dc/dc_mlx5.c

```diff
@@ -490,22 +490,23 @@ ucs_status_t uct_dc_mlx5_iface_dci_connect(uct_dc_mlx5_iface_t *iface,
     uct_ib_mlx5_md_t *md = ucs_derived_of(iface->super.super.super.super.md,
                                           uct_ib_mlx5_md_t);
     uct_ib_device_t *dev = uct_ib_iface_device(&iface->super.super.super);
-    uct_dc_mlx5_dci_config_t config;
+    uct_dc_mlx5_dci_config_t *config, temp_config;
     struct ibv_qp_attr attr;
     long attr_mask;
     ucs_status_t status;
 
     /* in case dci is not attached to a dci pool*/
     if (dci->pool_index != UCT_DC_MLX5_IFACE_INVALID_POOL_INDEX) {
-        config = iface->tx.dci_pool[dci->pool_index].config;
+        config = &iface->tx.dci_pool[dci->pool_index].config;
```

Comment:
can we avoid calling this function (uct_dc_mlx5_iface_dci_connect) if pool_index is invalid?
maybe can also remove UCT_DC_MLX5_IFACE_INVALID_POOL_INDEX define